lib/j_chars.rs

```diff
@@ -0,0 +1,139 @@
+#[derive(Debug)]
+pub struct JavaChars {
+	contents: Vec<u8>
+}
+
+impl JavaChars {
+	pub fn new(value: &str) -> JavaChars {
+		let mut vector:Vec<u8> = vec![];
+		for i in value.chars() {
+			let j:i32 = i as i32;
+			match i {
+				'\0' | '\u{80}' ... '\u{7FF}' => {
+					vector.push(b'\xC0' | (j >> 6) as u8)
+						; vector.push(0x80|(0x3F & j) as u8)
+				}
+				'\u{1}' ... '\u{7F}' => vector.push(i as u8),
+				'\u{800}' ... '\u{FFFF}' => {
+					vector.push(b'\xE0' | (j >> 12) as u8)
+						; vector.push(0x80 | (0x3F & (j >> 6)) as u8)
+						; vector.push(0x80 | (0x3F & j) as u8)
+				}
+				'\u{10000}' ... '\u{10FFFF}' => {
+					let subchar = j - 0x10000;
+					vector.push(0b11101101);
+					vector.push(0b10100000 | (subchar >> 16) as u8);
+					vector.push(0b10000000 | (0x3F & (subchar >> 10)) as u8);
+					vector.push(0b11101101);
+					vector.push(0b11101101);
+					vector.push(0b10100000 | (0x0F & (subchar >> 6)) as u8);
+					vector.push(0b10000000 | (0x3F & subchar) as u8);
+				}
+				_ => unreachable!()
+			}
+		}
+		vector.push(b'\0');
+		JavaChars { contents: vector, }
+	}
+	pub fn to_string(&self) -> Option<String> {
+		let mut a = Vec::<u8>::with_capacity(self.contents.len());
+		let mut counter:usize = 0;
+		loop {
```

Comment:
Are you sure, there are no standart decoders, or at least some library to do this?